### PR TITLE
provider/kubernetes: Upsert load balancer

### PR DESCRIPTION
**Pull request by lwander**
Implements upsert load balancer for kubernetes.

Load balancers are represented as kuberentes services (and can optionally be backed by a cloud provider's load balancer, when possible). 

Relationships between load balancers and instances (pods) are maintained using labels & selectors.

@duftler

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/converters/servergroup/CloneKubernetesAtomicOperationConverter.groovy [hunk 1]
```diff
@@ -1,11 +1,11 @@
 /*
- * Copyright 2016 Google, Inc.
+ * Copyright 2016 Netflix, Inc.
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/converters/servergroup/CloneKubernetesAtomicOperationConverter.groovy]: Whoops, just noticed the license headers were thrown off by IntelliJ. I'll update these in the morning.

**Comment by lwander** [comment]
@ttomsu

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/validators/StandardKubernetesAttributeValidator.groovy [hunk 2]
```diff
@@ -63,6 +78,46 @@ class StandardKubernetesAttributeValidator {
...
 
+  def validateProtocol(String value, String attribute) {
+    if (validateNotEmpty(value, attribute)) {
+      return validateByContainment(value, attribute, protocolList)
+    } else {
+      return false
+    }
+  }
+
+  def validateIpv4(String value, String attribute) {
```
Comment (ttomsu) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/validators/StandardKubernetesAttributeValidator.groovy]:
```
def result = InetAddressUtils.isIPv4Address(value)
if (!result) {
  errors.rejectValue(...)
}
result

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/validators/StandardKubernetesAttributeValidator.groovy [hunk 3]
```diff
@@ -63,6 +78,46 @@ class StandardKubernetesAttributeValidator {
...
+    if (InetAddressUtils.isIPv4Address(value)) {
+      result = true
+    } else {
+      errors.rejectValue("${context}.${attribute}", "${context}.${attribute}.invalid (Not valid IPv4 address)")
+      result = false
+    }
+    result
+  }
+
+  def validateServiceType(String value, String attribute) {
```
Comment (ttomsu) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/validators/StandardKubernetesAttributeValidator.groovy]:
```
value ? validateByContainment(value, attribute, serviceTypeList) : true
```

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/validators/StandardKubernetesAttributeValidator.groovy [hunk 4]
```diff
@@ -63,6 +78,46 @@ class StandardKubernetesAttributeValidator {
...
+    if (!value) {
+      result = true
+    } else {
+      result = validateByContainment(value, attribute, serviceTypeList)
+    }
+    result
+  }
+
+  def validatePort(int port, String attribute) {
+    def result
```
Comment (ttomsu) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/validators/StandardKubernetesAttributeValidator.groovy]: Same code shrinking as above.

**Comment by ttomsu** [comment]
LGTM with comments.

**Pull request merged**